### DISCUSSION TITLE
Indicate when response filter (JSONPath/XPath) input is invalid/non-existent

### DISCUSSION THREAD

**Discussion by yozlet** [Kong/insomnia#3707] [Ideas]
Given that the response filter is evaluated on every keypress, it should be possible to tell whether the JSONPath/XPath/whatever expression is in one of these three states while typing:

1. Incorrect/unprocessable syntax
2. Syntactically valid, but broken for the current document (e.g. addressing a path that doesn't exist in the doc, so returns `[]` or similar)
3. Useful (returns any nodes at all)

For my uses, I see states 1 & 2 as effectively identical in that I need to fix the filter input. Right now, the only indication that I'm in either state is that the Response Preview just shows `[]` or similar, and that's bad for the reasons discussed in #1259.

It would be great to have a a visual indicator or decoration for these states (e.g. changing text colour, or adding a red dotted underline, or changing the background colour), as well as making that state queryable by the rest of the app (for #1259, maybe others).

**Comment by welcome** [comment]
👋 Thanks for opening your first issue! If you're reporting a 🐞 bug, please make sure
you include steps to reproduce it. If you're requesting a feature 🎁, please provide real
use cases that would benefit. 👪

To help make this a smooth process, please be sure you have first read the
[contributing guidelines](https://github.com/getinsomnia/insomnia/blob/master/CONTRIBUTING.md).

**Comment by gschier** [comment]
I think it would definitely help to show syntax errors 👍
